packages/grid/_modules_/grid/hooks/core/preProcessing/useGridPreProcessing.ts

```diff
@@ -47,10 +46,6 @@ export const useGridPreProcessing = (apiRef: GridApiRef) => {
     [],
   );
 
-  const preProcessingApi: GridPreProcessingApi = {
-    unstable_registerPreProcessor: registerPreProcessor,
-    unstable_applyPreProcessors: applyPreProcessors,
-  };
-
-  useGridApiMethod(apiRef, preProcessingApi, 'GridPreProcessing');
+  apiRef.current.registerMethod('registerPreProcessor', false, registerPreProcessor);
```

Comment:
A variant could look something like 

```suggestion
  apiRef.current.registerMethods({
    registerPreProcessor: { method: registerPreProcessor, isPublic: false },
    applyPreProcessors: { method: applyPreProcessors, isPublic: false },
  });  
```

Or

```suggestion
  apiRef.current.registerMethods<{}, GridPrivatePreProcessor>({
    public: {},
    private: { registerPreProcessor, applyPreProcessors },
  });  
```

The last one is safer for full typing when we will have distinct interfaces for the free, the pro and the premium plan apis.
But it would make dynamic public / private uglier to register

```tsx
  const advancedEditingMethod = {  
    commitCellChange, 
    commitRowChange,
    setEditRowsModel,
    getEditRowsModel,
    setEditCellValue
  }

  const isAdvancedEditingApiPublic = props.signature !== GridSignature.DataGrid;
    
  apiRef.current.registerMethods({
    public: {
      setCellMode,
      getCellMode,
      getRowMode,
      isCellEditable,
      ...isAdvancedEditingApiPublic && advancedEditingMethod
    },
    private: { 
      registerPreProcessor,
      applyPreProcessors,
      ...!isAdvancedEditingApiPublic && advancedEditingMethod
    },
  });  
```